UI/DesignWnd.h

```diff
@@ -2,8 +2,49 @@
 #define _DesignWnd_h_
 
 #include <GG/Wnd.h>
+#include <boost/uuid/uuid.hpp>
 
 class EncyclopediaDetailPanel;
+class SaveGameUIData;
+
+struct Availability {
+    // Declaring an enum inside a struct makes the syntax when using the enum with tuples simpler,
+    // without polluting the global namespace with 3 generic names.
+
+    enum Enum {Obsolete,  // A design/part is researched/known by the player has marked it obsolete
+               Available, // A design/part is researched/known and currently available
+               Future     // A design/part is unresearched and hence not available
+    };
+};
+
+/** ShipDesignManager tracks information known to the client about ShipDesigns.  This includes
+    the order of designs in the UI. The information is currently used in the DesignWnd and the
+    BuildDesignator. */
+class ShipDesignManager {
+public:
+
+    /** Designs provides ordered lists of designs for display in the UI. */
+    class Designs {
+    public:
+        virtual std::vector<int> OrderedIDs() const
+            /* VC++ won't link ShipDesignManager if Designs is pure *///= 0;
+        { return std::vector<int>{}; }
+    };
+
+    ShipDesignManager();
+    ~ShipDesignManager();
+
+    void StartGame(int empire_id);
+    void Load(const SaveGameUIData& data);
```

Comment:
All changes are done.

Thanks a lot for figuring this out.  I was looking at the msvc project file for causes of the error.

Do you understand why VC++ requires it to be `virtual`?  It is never sub-classed.